### PR TITLE
Fix PDF delivery report

### LINKED ISSUE #3862: Delivery report pdf conversion off?
It seems the pdf conversion of delivery reports died. 

The pdf download button downloads html, much as if it had had `pdf_export: False`.
Screenshot 2023-03-21 at 10 18 37

But it doesn't, apparently: 
https://github.com/Clinical-Genomics/scout/blob/6a597662854de7f75d08420446f7fbe7ddaa94f8/scout/constants/case_tags.py#L5

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3862 
I am under the impression that the PDF version hasn't been working for a while

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Run scout setup demo and make sure that the case contains 2 new reports that can be seen/downloaded from the case sidebar -> pipeline_version and reference_info
2. Try to download the delivery report in the PDF format

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

File: scout/constants/case_tags.py
Comment (dnil) [inline scout/constants/case_tags.py:5]: 🙄

Comment (northwestwitch) [reply scout/constants/case_tags.py:5]: Ok, I'd rather keep that name and change the template, wait for the review

Comment (northwestwitch) [reply scout/constants/case_tags.py:5]: Naa it's more comoplicated. I'd like to keep it like this instead

**Comment by northwestwitch** [comment]
Yes, it has been broken since [Nov 22nd](https://github.com/Clinical-Genomics/scout/commit/aa235cdaede62285ffc8a490bd33ac5b0a21fde6), when [this code](https://github.com/Clinical-Genomics/scout/blob/79e171597bfdadea788832c763b510817b4855f7/scout/server/blueprints/cases/views.py#L728) was introduced.

**Comment by northwestwitch** [comment]
Ok, it should be fixed now. I wanted to fix the Vulture issues but I've realized that I can't: those --update are parameters that were present before in the cli and they aren't used any more, but if we remove them then it becomes a breaking change (major). We marked that cli function as deprecated for this reason.

**Comment by dnil** [comment]
> Yes, it has been broken since [Nov 22nd](https://github.com/Clinical-Genomics/scout/commit/aa235cdaede62285ffc8a490bd33ac5b0a21fde6), when [this code](https://github.com/Clinical-Genomics/scout/blob/79e171597bfdadea788832c763b510817b4855f7/scout/server/blueprints/cases/views.py#L728) was introduced.

Hm? But it should get this: https://github.com/Clinical-Genomics/scout/blob/63b99eba44ad8873673f34b8fe6b7a6c77bf3b83/scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html#L132, which should be the key_name to use?

**Review by dnil — APPROVED** [review]
This is fine; it solves the issue, and it would not be the first time we use explicit file endings (bam and cram files and their indexes come to mind). But see alternative solution to keep using the const in the other PR. I'll leave it to you!

**Comment by northwestwitch** [comment]
To better test the reports  (pipeline version and  reference_info ) I've included them in the demo config file, so they are loaded by default in the demo case. But I've noticed that the pipeline version one doesn't show on the case sidebar. It was due to a different naming of the key in the stored case object, now it is fixed!

**Comment by northwestwitch** [comment]
I see what do you mean for the test @dnil , I'll refactor to cover all the situations!

**Comment by dnil** [comment]
> To better test the reports (pipeline version and reference_info ) I've included them in the demo config file, so they are loaded by default in the demo case. But I've noticed that the pipeline version one doesn't show on the case sidebar. It was due to a different naming of the key in the stored case object, now it is fixed!

Yes, if I recall correctly we kept it at optional load since demand for them was limited, but agreed, why not load them when we have them anyway. 

I do not think we should change the actual config file key name or db field name of any of them - that's synced with other cg services and has been in rotation for a bit. But maybe I misunderstood, I'll check the code.

**Comment by northwestwitch** [comment]
> I do not think we should change the actual config file key name or db field name of any of them - that's synced with other cg services and has been in rotation for a bit. But maybe I misunderstood, I'll check the code.

I didn't change it (I think), the expected key was exe_ver and still is. But yes, take a look because it's a tricky one

File: scout/models/case/case.py
Comment (northwestwitch) [inline scout/models/case/case.py:57]: I think the key saved on the database for the case is pipeline_version, at least it's the one that is expected on the [formats constant](https://github.com/Clinical-Genomics/scout/blob/79e171597bfdadea788832c763b510817b4855f7/scout/constants/case_tags.py#L8)

File: tests/server/blueprints/cases/test_cases_views.py
Comment (northwestwitch) [inline tests/server/blueprints/cases/test_cases_views.py:14]: To make sure that this test is good I've tried reverting the code in cases_views and indeed the error is caught:

image

The tests fails with this message:

image

That happens when downloading the PDF report for the delivery reports -> instead of the PDF report you get an error page

File: scout/build/case.py
Comment (dnil) [inline scout/build/case.py:37]: Perhaps also add the `reference_info` then if we decided to load them systematically. They complement each other well for MIP?

File: scout/models/case/case.py
Comment (dnil) [reply scout/models/case/case.py:57]: But hold on, this is still weird: if this is the db key, then we have a bunch of other places to change it as well, don't we? Like https://github.com/Clinical-Genomics/scout/blob/b0cfc8795392ed7e1b223eeaa5ad5590fc6e8892/scout/adapter/mongo/case.py#L1000?

File: scout/build/case.py
Comment (dnil) [inline scout/build/case.py:37]:
```suggestion
def _assign_pipeline_info(case_obj, case_data):
```
Or just `build_pipeline_info` - its just language, but we've generally called the addition of data to the in memory objects for web rendering 'population' in contrast to the 'build' for converting the objects from memory for db storage, though there are some unfortunate reuses.. 😓

**Review by dnil — COMMENTED** [review]
Sorry, this is getting complicated, but thank you for adding the test fix; I think that was very relevant. The test would have caught the issue if it was right from the start, and it would also have made it more obvious what the different dictionary and database key names were.

File: scout/build/case.py
Comment (northwestwitch) [reply scout/build/case.py:37]: Yes the other key is already collected fine, because the case key has the same name that is taken from the config file

File: scout/models/case/case.py
Comment (northwestwitch) [reply scout/models/case/case.py:57]: Ok it looks weird but if you try the command then it updates the path correctly, perhaps we can change the comments in there to be less confusing

Comment (northwestwitch) [reply scout/models/case/case.py:57]: The cli command to update that field is not invoking that command it seems, but I fixed the adapter so that whenever you make another update in the case you keep also the pipeline_version

**Review by dnil — COMMENTED** [review]
Ok, looks good to me now, but I think let us test a bit with different loading and updating?

**Comment by northwestwitch** [comment]
Merging this one now

**Pull request merged**
